source/sigv4.c

```diff
@@ -2713,50 +2714,45 @@ static SigV4Status_t writeStringToSign( const SigV4Parameters_t * pParams,
                                         CanonicalContext_t * pCanonicalContext )
 {
     SigV4Status_t returnStatus = SigV4Success;
-    size_t encodedLen = pCanonicalContext->bufRemaining;
     char * pBufStart = ( char * ) pCanonicalContext->pBufProcessing;
     ptrdiff_t bufferLen = pCanonicalContext->pBufCur - pBufStart;
+    /* An overestimate but sufficient memory is checked before proceeding. */
+    size_t encodedLen = SIGV4_PROCESSING_BUFFER_LENGTH;
+    size_t sizeNeededBeforeHash = algorithmLen + 1U + \
```

Comment:
Could be helpful to comment what each part before the hash is. Are the 1s for newline characters?